### PR TITLE
Dry publication

### PR DESCRIPTION
**Pull request by SteelAlloy**
(forked from #74)

- In addition to the tests, eggs will now be test published.
- If an error arises during publication, `eggs-debug.log` is saved as an artifact.
- When publishing, the workflow will now resolve version from `version.ts`. It allows us to release unstable versions, eg. `2.0.0-beta_1`

### CODE REVIEW

**Comment by SteelAlloy** [comment]
I discover bugs that I would never have been able to find otherwise 😄
image

File: src/commands/publish.ts [hunk 1]
```diff
@@ -75,27 +75,6 @@ function ensureFiles(config: Config, matched: MatchedFile[]): boolean {
 }
 
 async function deprecationWarnings(config: Config) {
-  const name = config.name.toLowerCase();
-
-  try {
```
Comment (t8) [inline src/commands/publish.ts:80]: Hoping that this doesn't cause merge conflicts with #74

File: src/keyfile.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ function slugify(text: string) {
 
 export async function writeAPIKey(key: string): Promise<void> {
   const keyPath = join(envHOMEDIR(), KEY_FILE);
+  log.debug("Key path", keyPath);
```
Comment (t8) [inline src/keyfile.ts:23]: @oganexon Should this still be here?

Comment (SteelAlloy) [reply src/keyfile.ts:23]: @t8 Yes, by default it is not displayed 
You have to use the `--debug` option to the message

**Review by t8 — APPROVED** [review]
LGTM!

**Pull request merged**